Actually I could reproduce it locally on Fedora 42 with mingw64-gcc-14.2.1-4.fc42, I was just attempting to compile the wrong file last time I tried.

Here's how to reproduce it:
```
x86_64-w64-mingw32-g++ -o modules/gdscript/gdscript_tokenizer_buffer.windows.template_release.x86_64.o -c -std=gnu++17 -fno-exceptions -Wctor-dtor-privacy -Wnon-virtual-dtor -Wplacement-new=1 -Wvirtual-inheritance -ffp-contract=off -mwindows -fdiagnostics-color -O3 -Wall -Wextra -Wwrite-strings -Wno-unused-parameter -Wenum-conversion -Wshadow -Wno-misleading-indentation -Wno-return-type -Walloc-zero -Wduplicated-branches -Wduplicated-cond -Wstringop-overflow=4 -Wattribute-alias=2 -Wlogical-op -Werror -Wno-unknown-pragmas -DNDEBUG -DSTRICT_CHECKS -DWINDOWS_ENABLED -DWASAPI_ENABLED -DWINMIDI_ENABLED -DWINVER=0x0602 -D_WIN32_WINNT=0x0602 -DACCESSKIT_ENABLED -DVULKAN_ENABLED -DRD_ENABLED -DD3D12_ENABLED -DGLES3_ENABLED -DEGL_STATIC -DMINGW_ENABLED -DMINGW_HAS_SECURE_API=1 -DMINIZIP_ENABLED -DBROTLI_ENABLED -DTHREADS_ENABLED -DCLIPPER2_ENABLED -DZSTD_STATIC_LINKING_ONLY -DUSE_VOLK -DVK_USE_PLATFORM_WIN32_KHR -DGLAD_ENABLED -DEGL_ENABLED -DGODOT_MODULE -DTESTS_ENABLED -Iaccesskit-c-0.15.1/include -Iplatform/windows -I. -isystem thirdparty/libpng -isystem thirdparty/glad -isystem thirdparty/directx_headers/include/directx -isystem thirdparty/volk -isystem thirdparty/vulkan -isystem thirdparty/vulkan/include -isystem thirdparty/zstd -isystem thirdparty/zlib -isystem thirdparty/clipper2/include -isystem thirdparty/brotli/include -isystem thirdparty/angle/include -isystem thirdparty/d3d12ma modules/gdscript/gdscript_tokenizer_buffer.cpp
In file included from ./thirdparty/mingw-std-threads/mingw.mutex.h:58,
                 from ./core/os/mutex.h:37,
                 from ./core/os/thread_safe.h:33,
                 from ./core/object/message_queue.h:34,
                 from ./core/object/object.h:35,
                 from ./core/variant/binder_common.h:34,
                 from ./core/object/method_bind.h:33,
                 from ./core/object/class_db.h:33,
                 from ./core/object/ref_counted.h:33,
                 from ./core/io/marshalls.h:34,
                 from modules/gdscript/gdscript_tokenizer_buffer.cpp:34:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/errhandlingapi.h:43:26: error: variable 'GetErrorMode' definition is marked dllimport
   43 |   WINBASEAPI UINT WINAPI GetErrorMode (VOID);
      |                          ^~~~~~~~~~~~
/usr/x86_64-w64-mingw32/sys-root/mingw/include/errhandlingapi.h:43:40: error: 'VOID' was not declared in this scope; did you mean 'PVOID'?
   43 |   WINBASEAPI UINT WINAPI GetErrorMode (VOID);
      |                                        ^~~~
      |                                        PVOID
/usr/x86_64-w64-mingw32/sys-root/mingw/include/errhandlingapi.h:47:14: error: 'VOID' does not name a type; did you mean 'PVOID'?
   47 |   WINBASEAPI VOID WINAPI RaiseException (DWORD dwExceptionCode, DWORD dwExceptionFlags, DWORD nNumberOfArguments, CONST ULONG_PTR *lpArguments);
      |              ^~~~
      |              PVOID
/usr/x86_64-w64-mingw32/sys-root/mingw/include/errhandlingapi.h:49:27: error: variable 'GetLastError' definition is marked dllimport
   49 |   WINBASEAPI DWORD WINAPI GetLastError (VOID);
      |                           ^~~~~~~~~~~~
/usr/x86_64-w64-mingw32/sys-root/mingw/include/errhandlingapi.h:49:41: error: 'VOID' was not declared in this scope; did you mean 'PVOID'?
   49 |   WINBASEAPI DWORD WINAPI GetLastError (VOID);
      |                                         ^~~~
      |                                         PVOID
/usr/x86_64-w64-mingw32/sys-root/mingw/include/errhandlingapi.h:50:14: error: 'VOID' does not name a type; did you mean 'PVOID'?
   50 |   WINBASEAPI VOID WINAPI SetLastError (DWORD dwErrCode);
      |              ^~~~
      |              PVOID
```

--- a/core/string/string_name.cpp
+++ b/core/string/string_name.cpp
@@ -30,6 +30,7 @@
 
 #include "string_name.h"
 
+#include "core/os/mutex.h"
 #include "core/os/os.h"
 #include "core/string/print_string.h"
 
@@ -174,14 +175,6 @@ int StringName::length() const {
 	}
 
 	return 0;
-}
-
-bool StringName::is_empty() const {
-	if (_data) {
-		return _data->name.is_empty();
-	}
-
-	return true;
 }
 
 StringName &StringName::operator=(const StringName &p_name) {

--- a/core/string/string_name.h
+++ b/core/string/string_name.h
@@ -30,7 +30,6 @@
 
 #pragma once
 
-#include "core/os/mutex.h"
 #include "core/string/ustring.h"
 #include "core/templates/safe_refcount.h"
 
@@ -78,7 +77,7 @@ class StringName {
 	StringName(_Data *p_data) { _data = p_data; }
 
 public:
-	explicit operator bool() const { return _data && !_data->name.is_empty(); }
+	_FORCE_INLINE_ explicit operator bool() const { return _data; }
 
 	bool operator==(const String &p_name) const;
 	bool operator==(const char *p_name) const;
@@ -88,7 +87,7 @@ public:
 	const char32_t *get_data() const { return _data ? _data->name.ptr() : U""; }
 	char32_t operator[](int p_index) const;
 	int length() const;
-	bool is_empty() const;
+	_FORCE_INLINE_ bool is_empty() const { return !_data; }
 
 	_FORCE_INLINE_ bool is_node_unique_name() const {
 		if (!_data) {

--- a/modules/jolt_physics/spaces/jolt_contact_listener_3d.h
+++ b/modules/jolt_physics/spaces/jolt_contact_listener_3d.h
@@ -30,6 +30,7 @@
 
 #pragma once
 
+#include "core/os/mutex.h"
 #include "core/templates/hash_map.h"
 #include "core/templates/hash_set.h"
 #include "core/templates/hashfuncs.h"

